sys-apps/portage/portage-9999.ebuild

```diff
@@ -127,12 +127,17 @@ my_src_configure() {
 		-Ddocdir="${EPREFIX}/usr/share/doc/${PF}"
 		$(meson_use doc)
 		$(meson_use apidoc)
-		$(meson_use native-extensions)
 		$(meson_use gentoo-dev)
 		$(meson_use ipc)
 		$(meson_use xattr)
 	)
 
+	if use native-extensions && [[ "${EPYTHON}" != "pypy3" ]] ; then
```

Comment:
Does Portage actually have pure Python fallbacks for everything provided by the extensions?

Reply:
It does, yeah.

Reply:
One exception is that the check_locale function just returns None when the extension is disabled, which prevents a possible warning when the LC_CTYPE locale is not posixish.

Reply:
```
$ pypy3.10 -c "from portage.util.locale import check_locale; print(check_locale())"
True

$ python3.12 -c "from portage.util.locale import check_locale; print(check_locale())"
True

$ qlist portage | grep "\.so"
/usr/lib/portage/pypy3/ebuild-helpers/newlib.so
/usr/lib/portage/pypy3/ebuild-helpers/dolib.so
/usr/lib/portage/python3.11/ebuild-helpers/newlib.so
/usr/lib/portage/python3.11/ebuild-helpers/dolib.so
/usr/lib/portage/python3.12/ebuild-helpers/newlib.so
/usr/lib/portage/python3.12/ebuild-helpers/dolib.so
/usr/lib/python3.11/site-packages/portage/util/libc.cpython-311-x86_64-linux-gnu.so
/usr/lib/python3.11/site-packages/portage/util/_whirlpool.cpython-311-x86_64-linux-gnu.so
/usr/lib/python3.11/site-packages/portage/util/file_copy/reflink_linux.cpython-311-x86_64-linux-gnu.so
/usr/lib/python3.12/site-packages/portage/util/libc.cpython-312-x86_64-linux-gnu.so
/usr/lib/python3.12/site-packages/portage/util/_whirlpool.cpython-312-x86_64-linux-gnu.so
/usr/lib/python3.12/site-packages/portage/util/file_copy/reflink_linux.cpython-312-x86_64-linux-gnu.so
```

Reply:
I'm still not sure. I don't think the code in question is performance critical, and I think C extensions are more reliable than `ctypes` in the long run.

Reply:
Portage uses it for every file copy, not just at merge-time, and the context-switch is _very_ expensive in pypy.